subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompilerFactory.java

```diff
@@ -28,25 +28,40 @@ public static TwirlCompiler create(PlayPlatform playPlatform) {
     public static VersionedTwirlCompilerAdapter createAdapter(PlayPlatform playPlatform) {
         String playVersion = playPlatform.getPlayVersion();
         String scalaCompatibilityVersion = playPlatform.getScalaPlatform().getScalaCompatibilityVersion();
+        VersionedPlayTwirlAdapter playTwirlAdapter = createPlayTwirlAdapter(playPlatform);
         switch (PlayMajorVersion.forPlatform(playPlatform)) {
             case PLAY_2_2_X:
-                return new TwirlCompilerAdapterV22X("2.2.6", scalaCompatibilityVersion);
+                return new TwirlCompilerAdapterV22X("2.2.6", scalaCompatibilityVersion, playTwirlAdapter);
             case PLAY_2_3_X:
-                return new TwirlCompilerAdapterV10X("1.0.4", scalaCompatibilityVersion);
+                return new TwirlCompilerAdapterV10X("1.0.4", scalaCompatibilityVersion, playTwirlAdapter);
             case PLAY_2_4_X:
             case PLAY_2_5_X:
-                return new TwirlCompilerAdapterV10X("1.1.1", scalaCompatibilityVersion);
+                return new TwirlCompilerAdapterV10X("1.1.1", scalaCompatibilityVersion, playTwirlAdapter);
             case PLAY_2_6_X:
                 // TODO: create a TwirlJavaCompiler for Twirl 1.3.x that uses the Java compiler interface
                 // Rename TwirlCompiler to TwirlScalaCompiler
-                // Waiting for https://github.com/playframework/twirl/pull/136
-                // and https://github.com/gradle/gradle/pull/2062
                 // We don't want to use the legacy TwirlScalaCompiler for Twirl 3.0 because the interface is more
                 // complicated and it would be really frustrating to deal with typed parameters via reflection,
                 // to create Scala Seqs in Java, etc.
-                return new TwirlCompilerAdapterV10X("1.1.1", scalaCompatibilityVersion);
+                return new TwirlCompilerAdapterV10X("1.1.1", scalaCompatibilityVersion, playTwirlAdapter);
```

Comment:
Play 2.6 uses Twirl 1.3.x series:

https://github.com/playframework/playframework/blob/2.6.x/framework/project/plugins.sbt#L14

Reply:
Also, I think it needs to use `TwirlCompilerAdapterV22X` instead of `TwirlCompilerAdapterV10X`.

Reply:
Yes. I was just refactoring for now. Adding support for Twirl 1.3.x was going to be my next PR after this one is merged

V22X is for the version of Twirl bundled with Play 2.2. V10X is for the standalone version of Twirl. It's a bit confusing. I'm hoping we can just delete all the V22X stuff in a follow up PR because I don't know that we still need to support such an old version of Play

Reply:
Thanks for explaining this, @benmccann.

I think adding some comments could help to make this clearer. I was looking for some comment about why we have `TwirlCompilerAdapterV22X` and then `TwirlCompilerAdapterV10X` since this is, as you said, confusing.

This could be beyond this PR though.